opm/input/eclipse/Schedule/BCProp.hpp

```diff
@@ -0,0 +1,114 @@
+/*
+  Copyright 2020 Equinor ASA.
+  Copyright 2023 Norce.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_BC_PROP_HPP
+#define OPM_BC_PROP_HPP
+
+#include <vector>
+#include <cstddef>
+#include <optional>
+
+#include <opm/input/eclipse/EclipseState/Grid/FaceDir.hpp>
+#include <opm/input/eclipse/EclipseState/Grid/GridDims.hpp>
+
+
+namespace Opm {
+
+class Deck;
+class DeckRecord;
+
+enum class BCType {
+     RATE,
+     FREE,
+     DIRICHLET,
+     THERMAL,
+     CLOSED,
+     NONE
+};
+
+enum class BCComponent {
+     OIL,
+     GAS,
+     WATER,
+     SOLVENT,
+     POLYMER,
+     NONE
+};
+
+
+class BCPROP {
```

Comment:
Class name should be BCProp as the file name, not BCPROP. It also creates confusion with the class generated from the JSON description, which also will be BCPROP I think.